Dart/testData/generate/EqualsAndHashCode1.txt

```diff
@@ -5,18 +5,18 @@ class EqualsAndHashCode1 {
   static var s;
 
   @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) {
-      return true;
-    }
-    return other is EqualsAndHashCode1 &&
-        this.str == other.str &&
-        this.v == other.v &&
-        this.i == other.i;
-  }
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+          other is EqualsAndHashCode1 &&
+              str == other.str &&
+              v == other.v &&
+              i == other.i;
 
   @override
-  int get hashCode {
-    return str.hashCode ^ v.hashCode ^ i.hashCode;
-  }<caret>
+  int get hashCode =>
+      super.hashCode ^
```

Comment:
Agree, keeping `other is ThisClass` to give analyzer a hint sounds reasonable.

Quick glance at [this article](http://www.angelikalanger.com/Articles/JavaSolutions/SecretsOfEquals/Equals-2.html) made me think that we are safer with `runtimeType == other.runtimeType`.